src/components/core/DpUpload/DpUpload.vue

```diff
@@ -95,6 +95,60 @@ export default {
       return b ? b.pop() : ''
     },
 
+    /**
+     * Fires immediately before a file is added to the Uppy store.
+     *
+     * The hook is used here to clean file names from characters that would
+     * break the tus endpoint. This could as well be implemented within the backend.
+     *
+     * @param currentFile
+     * @param files
+     * @return {(*&{meta: (*&{name: *}), name: *})|*}
+     * @see https://github.com/transloadit/uppy/blob/main/packages/@uppy/core/src/Uppy.js#L503
+     */
+    handleOnBeforeFileAdded (currentFile, files) {
+      let fileName = currentFile.name
+
+      /*
+       * This is not an exhaustive list - the characters have been determined
+       * simply by trying out which break or do not break the tus endpoint.
+       */
+      const reservedCharacters = [
+        '?',
+        '#',
+        '&',
+        '\'',
+        '"',
+        '/',
+        ':',
+        '\\'
+      ]
+
+      /*
+       * For reasons i could not figure out, the for loop was needed here
+       * to reliably access the item that is looped over. When using forEach,
+       * strangely it always held the value of currentFile.
+       */
+      for (let i = 0; i < reservedCharacters.length; i++) {
+        if (fileName.includes(reservedCharacters[i])) {
+          fileName = fileName.replace(reservedCharacters[i], '_')
+        }
+      }
+
+      if (fileName !== currentFile.name) {
+        return {
+          ...currentFile,
+          name: fileName,
+          meta: {
+            ...currentFile.meta,
+            name: fileName
```

Comment:
what exactly happens with the meta-infos?

Reply:
> what exactly happens with the meta-infos?

I had to change the name also within them, otherwise it did not work out. I assume the backend logic someway uses the meta infos when actually creating the file.